ls-alg/main/Main.java

```diff
@@ -0,0 +1,162 @@
+		RealMatrix matrix = new Array2DRowRealMatrix(numOfraws,numOfProfessions);
+		
+		int row = 0;
+		for(int i =0;i<numOfProfessions;i++)
+			for(int j =i+1;j<numOfProfessions;j++) {
+				matrix.addToEntry(row, i, 1);
+				matrix.addToEntry(row, j, -1);
+				row++;
+			}
+		return matrix.getData();
+		
+		
+	}
+	
+	public static double[][] generateSyntethicC(RealMatrix m ,int i, RealMatrix a){
+		RealMatrix c = a.multiply(m.getRowMatrix(i).transpose());
+		return c.getData();
+	}
+	
+	public static RealVector solve(RealMatrix a , RealVector c) {
+		
+		SingularValueDecomposition s = new SingularValueDecomposition(a);
+		DecompositionSolver d = s.getSolver();
+		return d.solve(c);
+		}
+	
+	public static double[][] buildCheck(int numOfProfessions,RealMatrix cMatrix,int numOfCharacteristics) {
+		RealMatrix a = new Array2DRowRealMatrix();
+		a = MatrixUtils.createRealMatrix(generateSyntheticA(numOfProfessions));
+		RealMatrix m = new Array2DRowRealMatrix(numOfCharacteristics,numOfProfessions);
+		
+		for(int i =0;i<numOfCharacteristics;i++) {
+			
+			RealVector c = cMatrix.getRowVector(i);
+			RealVector rowInM = solve(a,c);
+			Abs av = new Abs();
+			double min = av.value(rowInM.getMinValue());
+			rowInM.mapAddToSelf(min);
+			double max = av.value(rowInM.getMaxValue());
+			rowInM.mapDivideToSelf(max);
+			m.setRowVector(i, rowInM);
+		}
+		
+		return m.getData();
+	}
+	
+	//we read the matrix from csv file.
+	//we ignore the alphabet chars and build the matrix without them.
+	public static RealMatrix readMatrixAndCreate() {
+		
+		 Scanner scanner;
+		 StringBuilder b = new StringBuilder();
+		 RealMatrix rMatrix = new Array2DRowRealMatrix(3,3);
+		 Path currentRelativePath = Paths.get("");
+		 String myPath = currentRelativePath.toAbsolutePath().toString();
+		 
+		try {
+			scanner = new Scanner(new File(myPath +"/ls-alg/check.csv"));
+			scanner.useDelimiter(",");
+			//build one string of all the matrix
+			while(scanner.hasNext()) {
+				b.append(scanner.next() +" ");
+			}
+			//build string array,every string is a vector of the matrix
+			String[] s = b.toString().split("\r\n");
+	        int row = 0;
+	        for (String string : s) {
+	        	
+				String[] vectorToSet = string.split(" ");
+				//ignorance of alphabet chars vector
+				if(vectorToSet[1].charAt(0) >= 'a'
+						&& vectorToSet[1].charAt(0) <= 'z' ) {
+					continue;
+				}else {
+					
+					double[] data = new double[vectorToSet.length];
+					int i = 0;
+					for (String val : vectorToSet) {	
+						data[i]=(Double.parseDouble(val));
+						i++;
+					}
+					RealVector v = MatrixUtils.createRealVector(data);
+					rMatrix.setRowVector(row, v);	
+					
+				}
+				row++;	
+			}
+	        
+	 	} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		//the final matrix after build
+		return rMatrix;
+	}
+	
+	
+	
+	public static void main(String[] args) {
+		
+		
+		//read matrix from file and build
```

Comment:
Comments should start with a capital letter and a space, so for example here:
`//read`
would change to:
`// Read`